kncaputo/rancid-tomatillos
# Feature/initial testing

## What's this PR do?
- Add apiCall functions for single and all movies
- Wrote tests for all class and function components, including async (App.test.js)
- Error message handling for status code 400+
- Fix back button display/functionality (remove on home page view)
- Refactor conditional rendering for MovieContainer vs. MovieDetails
  - via adding isMovieDetails: true/false property to state (App.js)

## Where should the reviewer start?
- Test & Component files for: MovieCard/MovieDetails/MovieContainer/App

## How should this be manually tested?
- run `npm test`, see that tests pass 
- run `npm start` and view on localhost: 3000.

## Any background context you want to provide?
- Still need to add sad path tests for error handling 

## Questions: @letakeane 
- Should we create functional components for ErrorMessage and Header/Nav?
- Does the way we grabbed the request status codes and set them to state make sense? Prior to doing that, when we forced a server error and printed the message, we noticed that the error being displayed was not related to the actual server error, which was the root of the error. Instead, the error that would display was a type or syntax error when our display was trying to map the data or access a property on the data. By accessing the status code we were able to get more information about what the actual error is. Since doing this though, none of the errors we've forced have resulted in a message. 
   - Is the catch still doing something the way we have it set up? 
   - Does what we did make sense?
   - Is there a better way of retrieving the status codes?
   - See `apiCalls.js` line 3 and 8, and `App.js` `componentDidMount` and `getSingleMovie`
